lib/facility_factory.rb

```diff
@@ -0,0 +1,37 @@
+class FacilityFactory
+
+
+  def initialize
+
+  end
+
+  def create_facilities(data)
+    new_facilities = data.map do |facility|
+      Facility.new(
+        {
+        name: facility[:title] || facility[:office_name] || facility[:name],
+
+        address: 
+        if !facility[:location_1].nil?
+          formatted_address = JSON.parse(facility[:location_1][:human_address], symbolize_names: true)
+          "#{formatted_address[:address]}, #{formatted_address[:city]}, #{formatted_address[:state]} #{formatted_address[:zip]}"
+        elsif !facility[:street_address_line_1].nil?
+          "#{facility[:street_address_line_1]}, #{facility[:city]}, #{facility[:state]} #{facility[:zip_code]}"
+        elsif !facility[:address1].nil?
+          "#{facility[:address1]}, #{facility[:city]}, #{facility[:state]} #{facility[:zipcode]}"
+        end, 
+        
+        phone: facility[:phone_number] || facility[:phone] ||= "N/A",
+        services: [],
+        registered_vehicles: [],
+        collected_fees: 0
+        }
+      )
+    end
+    
+  end
```

Comment:
This method is doing a lot. It's determining how to format data based on source, mapping and then creating facilities. I would like to see some helpers used here to make it a little easier to test and a whole lot easier to read:
```ruby
  def create_facilities(data)
    data.map do |facility|
      Facility.new(
        {
        name: format_name(facility),
        address: format_address(facility),
        phone: format_phone(facility),
        services: [],
        registered_vehicles: [],
        collected_fees: 0
        }
      )
    end
  end

  def format_name(facility)
    facility[:title] || facility[:office_name] || facility[:name]
  end

  def format_address(facility)
    if !facility[:location_1].nil?
      formatted_address = JSON.parse(facility[:location_1][:human_address], symbolize_names: true)
      "#{formatted_address[:address]}, #{formatted_address[:city]}, #{formatted_address[:state]} #{formatted_address[:zip]}"
    elsif !facility[:street_address_line_1].nil?
      "#{facility[:street_address_line_1]}, #{facility[:city]}, #{facility[:state]} #{facility[:zip_code]}"
    elsif !facility[:address1].nil?
      "#{facility[:address1]}, #{facility[:city]}, #{facility[:state]} #{facility[:zipcode]}"
    end
  end

  def format_phone(facility)
    facility[:phone_number] || facility[:phone] ||= "N/A"
  end
```